### DISCUSSION TITLE
How to log the solving-time in a csv format

### DISCUSSION THREAD

**Discussion by akseljohan** [casadi/casadi#3155] [Q&A]
I am using Casadi in an Optimal Control problem and want to log the time it takes to find a solution.  I tried to use the ipopt options: output_file: 'log.csv', file_print_level: 1,  print_timing_statistics: 'yes'. However, this gives an output that is more in a "text report" format and is not parsable. In the end, I want a plot to show the time it takes for the IPOPT to find a solution for each controller step.

Does anyone know about an alternative for logging the time it takes to find a solution by utilizing native Casadi functionality or IPOPT functionality? Since the data already exists (it is printed to the console and to the output file) I would prefer to not use additional overlay such as "Timeit" for example.

**Comment by jgillis** [comment]
There is nothing in CasADi built in, but you may obtain statistics as a struct in Matlab or dictionary in Python, and from there go to csv.


For example
```python
solver = nlpsol('solver','ipopt',nlp)
solver(x0=...)
stats = solver.stats()
```

or
```python
opti = Opti();
...
sol = opti.solve();
stats = sol.stats()
```
